NInce_DB_Week3

NesibeInce opened the pull request.

burakcoskun commented on Week3/homework/skills.js:
```diff
@@ -0,0 +1,83 @@
+);
+connection.query(
+  'CREATE TABLE IF NOT EXISTS language_skills (emp_no INT, mother_tongue VARCHAR(25), second_language VARCHAR(25), third_language VARCHAR(25), PRIMARY KEY (emp_no))',
+);
```
better to have `emp_no` in this tables as foreign keys, so that we can have multiple entries like `emp_no=1` and `backend=nodejs`, `emp_no=1` and `backend=golang` etc. or we can have a table that only has skills e.g.`skill_name` and `skill_id` and we can have another table with (`emp_no`,  `skill_id`) which relates employees with skills

burakcoskun commented on Week3/homework/skills.js:
```diff
@@ -0,0 +1,83 @@
+console.log(languageList);
+
+function skillQuery(arr) {
+  var query = `INSERT INTO employee SET ?`;
```
looks good, structured and clean coding

burakcoskun commented on Week3/homework/transactions.js:
```diff
@@ -0,0 +1,25 @@
+    const managerName = await connection.query(`SELECT manager FROM departments WHERE dept_no = ?`, dept_no);
+    const newReportsTo = await connection.query(`SELECT emp_name FROM employee WHERE emp_no = ?`, emp_no);
+    await connection.query(`UPDATE employee SET reports_to = ${newReportsTo} WHERE reports_to = ${managerName}`);
+    await connection.query(`COMMIT`);
```
looks good to me, correct usage of transactions

NesibeInce commented:
Thank you very much for your time and concern. It really motivates and
encourages me.



On Mon, Jan 27, 2020 at 9:22 PM Burak Coskun <notifications@github.com>
wrote:

> *@burakcoskun* approved this pull request.
> ------------------------------
>
> In Week3/homework/skills.js
> <https://github.com/HackYourHomework/databases/pull/33#discussion_r371461804>
> :
>
> > +connection.query(
> +  'CREATE TABLE IF NOT EXISTS technical_skills (emp_no INT, backend VARCHAR(25), frontend VARCHAR(25), other VARCHAR(25), PRIMARY KEY (emp_no))',
> +);
> +connection.query(
> +  'CREATE TABLE IF NOT EXISTS soft_skills (emp_no INT, team_work INT, communication INT, problem_solving INT, PRIMARY KEY (emp_no))',
> +);
> +connection.query(
> +  'CREATE TABLE IF NOT EXISTS language_skills (emp_no INT, mother_tongue VARCHAR(25), second_language VARCHAR(25), third_language VARCHAR(25), PRIMARY KEY (emp_no))',
> +);
>
> better to have emp_no in this tables as foreign keys, so that we can have
> multiple entries like emp_no=1 and backend=nodejs, emp_no=1 and
> backend=golang etc. or we can have a table that only has skills e.g.
> skill_name and skill_id and we can have another table with (emp_no,
> skill_id) which relates employees with skills
> ------------------------------
>
> In Week3/homework/skills.js
> <https://github.com/HackYourHomework/databases/pull/33#discussion_r371462357>
> :
>
> > +var techList = [];
> +
> +for (i in table.technical_skills) {
> +  var row = table.technical_skills[i].split(',');
> +  var newObj = {
> +    emp_no: row[0],
> +    backend: row[1],
> +    frontend: row[2],
> +    other: row[3],
> +  };
> +  techList.push(newObj);
> +}
> +
> +var softList = [];
> +
> +for (i in table.soft_skills) {
> +  var row = table.soft_skills[i].split(',');
> +  var newObj = {
> +    emp_no: row[0],
> +    team_work: row[1],
> +    communication: row[2],
> +    problem_solving: row[3],
> +  };
> +  softList.push(newObj);
> +}
> +
> +var languageList = [];
> +
> +for (i in table.language_skills) {
> +  var row = table.language_skills[i].split(',');
> +  var newObj = {
> +    emp_no: row[0],
> +    mother_tongue: row[1],
> +    second_language: row[2],
> +    third_language: row[3],
> +  };
> +  languageList.push(newObj);
> +}
> +
> +console.log(softList);
> +console.log(techList);
> +console.log(languageList);
> +
> +function skillQuery(arr) {
> +  var query = `INSERT INTO employee SET ?`;
>
> looks good, structured and clean coding
> ------------------------------
>
> In Week3/homework/transactions.js
> <https://github.com/HackYourHomework/databases/pull/33#discussion_r371462579>
> :
>
> > +    await connection.connect();
> +    await connection.query(`START TRANSACTION`);
> +    const managerName = await connection.query(`SELECT manager FROM departments WHERE dept_no = ?`, dept_no);
> +    const newReportsTo = await connection.query(`SELECT emp_name FROM employee WHERE emp_no = ?`, emp_no);
> +    await connection.query(`UPDATE employee SET reports_to = ${newReportsTo} WHERE reports_to = ${managerName}`);
> +    await connection.query(`COMMIT`);
>
> looks good to me, correct usage of transactions
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/HackYourHomework/databases/pull/33?email_source=notifications&email_token=AMQ5OODIT4XBN64SRN7P7QLQ7467PA5CNFSM4KIPH2M2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCTGIPSA#pullrequestreview-348948424>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMQ5OOB2QD5X63EBMOL43LTQ7467PANCNFSM4KIPH2MQ>
> .
>
